drinfomon.lic

```diff
@@ -815,11 +815,13 @@ while line = script.gets
         .select { |_town, titles| titles.include?(checkroom) }
         .each do |_town, _titles|
           contents = Regexp.last_match(1)
-          if line =~ /a (\w+ shelf|rack|box|case|tree) with some stuff (on|in) it/
-            fput 'look ' + Regexp.last_match(2) + ' ' + Regexp.last_match(1)
-            if get =~ /(?:On|In) the \w+ (?:shelf|rack|box|case|tree) you see(.*)\./
-              contents += Regexp.last_match(1)
-            end
+          if line =~ /a .+?\s?(?:(\w+ (?:shelf|rack|box|case|tree))) with some stuff (on|in) it/
+            line.scan(/a .+?\s?(?:(\w+ (?:shelf|rack|box|case|tree))) with some stuff (on|in) it/).each{|noun,location|
```

Comment:
💡 I think you no longer need the `if`, since `scan` will return an empty array if nothing matches.